File: 169-majority-element/majority-element.cpp
#include <unordered_map>
#include <vector>
using namespace std;
class Solution {
public:
    int majorityElement(vector<int>& nums) {
    unordered_map<int, int> freqMap;

    int n = nums.size();
    for (int num : nums) {
        freqMap[num]++; // Count the occurrences of each element
    }

    for (const auto& pair : freqMap) {
        if (pair.second > n / 2) { // Check if the frequency exceeds n/2
            return pair.first; // Return the majority element
        }
    }

    return -1; // Return -1 if no majority element exists
}
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {

        HashMap<Integer, Integer> hm = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (hm.containsKey(nums[i])) {
                hm.put(nums[i], hm.get(nums[i]) + 1);
            } else {
                hm.put(nums[i], 1);
            }
        }

        for (Map.Entry<Integer, Integer> entry : hm.entrySet()) {
            if (entry.getValue() > nums.length / 2) {
                return entry.getKey();
            }
        }
        return 0;
    }
}